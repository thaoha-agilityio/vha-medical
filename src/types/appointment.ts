import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export type AppointmentStatus = 0 | 1 | 2 | 3;

export interface AppointmentStatusOption {
  key: string;
  label: string;
  value: AppointmentStatus;
}

export interface AppointmentModel {
  startTime: string;
  durationTime: string;
  status: AppointmentStatus;
  senderId: APIRelatedResponse<APIResponse<UserModel>>;
  receiverId: APIRelatedResponse<APIResponse<UserModel>>;
}

export type AppointmentResponse = APIResponse<AppointmentModel>;

export type AppointmentDataResponse = {
  appointment: AppointmentResponse | null;
  error: string | null;
};

export type AppointmentsDataResponse = Promise<
  { appointments: AppointmentResponse[]; error: string | null } & MetaResponse
>;

export type AppointmentsResponse = {
  data: AppointmentResponse[];
  meta: MetaResponse;
};

export type AppointmentPayload = Partial<
  Omit<AppointmentModel, 'senderId' | 'receiverId'>
> & {
  senderId?: string;
  receiverId?: string;
};
