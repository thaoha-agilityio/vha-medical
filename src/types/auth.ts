import { UserSession } from './user';

export interface LoginFormData {
  identifier: string;
  password: string;
  remember: boolean;
}

export interface SignupFormData {
  username: string;
  email: string;
  password: string;
  confirmPassWord: string;
}

export type AuthResponse = {
  jwt: string;
  user: UserSession | null;
  error: string | null;
};

export interface RolePermission {
  id: number;
  name: string;
}

export type RolesResponse = {
  roles: RolePermission[];
  error: string | null;
};
