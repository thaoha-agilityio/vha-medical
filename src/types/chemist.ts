import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export type ChemistModel = {
  users_permissions_user: APIRelatedResponse<APIResponse<UserModel>>;
};

export type ChemistResponse = APIResponse<ChemistModel>;

export type ChemistDataResponse = {
  chemist: ChemistResponse | null;
  error: string | null;
};

export type ChemistsDataResponse = Promise<
  { chemists: ChemistResponse[]; error: string | null } & MetaResponse
>;

export type ChemistsResponse = {
  data: ChemistResponse[];
  meta: MetaResponse;
};

export type ChemistPayload = {
  users_permissions_user: string;
};

export interface ChemistFormData {
  username: string;
  email: string;
  password: string;
  confirmPassWord: string;
  avatar?: string;
  description?: string;
  rating?: number;
  tasks?: number;
  reviews?: number;
  specialtyId: string;
}
