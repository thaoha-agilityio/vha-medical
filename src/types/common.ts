// Components
export type CustomClassType = {
  customClass?: string;
};

export type PageErrorProps = {
  error: Error & { digest?: string };
  reset: () => void;
};

// Status
export enum STATUS_TYPE {
  SUCCESS = 'success',
  WARNING = 'warning',
  ERROR = 'error',
}

// Theme mode
export enum THEME_MODE_TYPE {
  DARK = 'dark',
  LIGHT = 'light',
}

// Option
export interface Option {
  key: string;
  label: string;
}

// FetchData
export interface FetchDataProps {
  searchParams?: URLSearchParams;
  options?: RequestInit;
}
