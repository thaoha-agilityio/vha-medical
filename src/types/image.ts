export type ImageExtension = 'webp' | 'png';

export type ImageInfo = {
  filename: string;
  name: string;
  mime: string;
  extension: string;
  url: string;
};

export type ImageData = {
  id: string;
  title: string;
  url_viewer: string;
  url: string;
  display_url: string;
  width: string;
  height: string;
  size: string;
  time: string;
  expiration: string;
  image: ImageInfo;
  thumb: ImageInfo;
  medium: ImageInfo;
  delete_url: string;
};

export type ImgBBResponse = {
  data: ImageData;
  success: boolean;
  status: number;
};
