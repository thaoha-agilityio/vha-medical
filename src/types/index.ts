export * from './appointment';
export * from './auth';
export * from './chemist';
export * from './common';
export * from './image';
export * from './notification';
export * from './response';
export * from './searchParams';
export * from './specialty';
export * from './table';
export * from './user';
