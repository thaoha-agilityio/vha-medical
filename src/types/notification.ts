import {
  APIRelatedResponse,
  APIResponse,
  AppointmentStatus,
  MetaResponse,
  UserModel,
} from '@/types';

export interface InfoModel {
  id: string;
  status: AppointmentStatus;
  startTime: string;
  durationTime: string;
  content: string;
}

export interface NotificationModel {
  senderName: string;
  senderAvatar: string;
  isRead: boolean;
  info: InfoModel;
  senderId?: APIRelatedResponse<APIResponse<UserModel>>;
  createdAt?: string;
}

export type NotificationResponse = APIResponse<NotificationModel>;

export type NotificationDataResponse = {
  notification: NotificationResponse | null;
  error: string | null;
};

export type NotificationsDataResponse = Promise<
  { notifications: NotificationResponse[]; error: string | null } & MetaResponse
>;

export type NotificationsResponse = {
  data: NotificationResponse[];
  meta: MetaResponse;
};

export type NotificationPayload = Omit<NotificationModel, 'senderId'> & {
  senderId: string;
};
