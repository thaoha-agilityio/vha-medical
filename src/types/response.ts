export type APIResponse<T> = { id: string; attributes: T };

export type APIRelatedResponse<T> = { data: T };

export type MetaResponse = {
  pagination?: {
    page: number;
    pageSize: number;
    pageCount: number;
    total: number;
  };
};

export type TDataResponse<T> = {
  data: APIResponse<T>[];
  meta?: MetaResponse;
};

export interface ErrorResponse {
  data?: null;
  error: {
    status?: number;
    name?: string;
    message: string;
    details?: Record<string, unknown>;
  };
}
