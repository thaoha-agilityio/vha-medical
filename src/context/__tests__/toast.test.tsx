// Libs
import { render, fireEvent, waitFor } from '@testing-library/react';
import React from 'react';

// Contexts
import ToastProvider, { useToast } from '@/context/toast';
import { STATUS_TYPE } from '@/types';

// Component that uses the toast context
const ChildWithOpenToast = () => {
  const openToast = useToast();

  const handleOpenToast = () => {
    openToast({
      type: STATUS_TYPE.SUCCESS,
      message: 'Test open toast success',
    });
  };

  return (
    <div>
      Test Toast Context
      <button onClick={handleOpenToast}>Open Toast</button>
    </div>
  );
};

// Test Suite
describe('ToastProvider and useToast', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const setupRenderWithOpenToast = () => {
    return render(
      <ToastProvider>
        <ChildWithOpenToast />
      </ToastProvider>,
    );
  };

  it('should render and open toast correctly and display the content message', async () => {
    const { container, getByText, queryByTestId } = setupRenderWithOpenToast();

    // Snapshot test
    expect(container).toMatchSnapshot();

    const button = getByText('Open Toast');
    fireEvent.click(button);

    // Verify toast visibility and content
    await waitFor(() => {
      expect(queryByTestId('toast')).toBeTruthy();
      expect(getByText('Test open toast success')).toBeTruthy();
    });
  });
});
