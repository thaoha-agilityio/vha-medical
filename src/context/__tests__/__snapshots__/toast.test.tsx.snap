// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastProvider and useToast should render and open toast correctly and display the content message 1`] = `
<div>
  <div>
    Test Toast Context
    <button>
      Open Toast
    </button>
  </div>
</div>
`;
